### PR TITLE
PLT-798 Prevent image files larger than 4k resolution from being uploaded

### PR DESCRIPTION
**Pull request by hmhealey**

### CODE REVIEW

File: api/file.go [hunk 1]
```diff
@@ -125,6 +125,21 @@ func uploadFile(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		if model.IsFileExtImage(filepath.Ext(files[i].Filename)) {
+			imageNameList = append(imageNameList, uid+"/"+filename)
+			imageDataList = append(imageDataList, buf.Bytes())
+
+			// Decode image config first to check dimensions before loading the whole thing into memory later on
+			config, _, err := image.DecodeConfig(bytes.NewReader(buf.Bytes()))
+			if err != nil {
+				c.Err = model.NewAppError("uploadFile", "Unable to upload image file.", err.Error())
+				return
+			} else if config.Width*config.Height > 4096*2160 {
```
Comment (jwilander) [inline api/file.go]: `4096*2160` is used in multiple places, maybe make it a constant?

**Comment by jwilander** [comment]
+1, other than my one comment

**Comment by crspeller** [comment]
+1

**Comment by coreyhulen** [comment]
+1 I'm going to go ahead and merge this, but I think the 50Mb limit blocking img uploads also needs to be ported to upload profile pic.

**Pull request merged**